src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/ThrowsTaglet.java

```diff
@@ -196,34 +197,38 @@ private Content inheritThrowsDocumentation(ExecutableElement holder,
                                                Set<String> alreadyDocumented,
                                                Map<String, TypeMirror> typeSubstitutions,
                                                TagletWriter writer) {
-        Utils utils = writer.configuration().utils;
         Content result = writer.getOutputInstance();
-        if (utils.isMethod(holder)) {
-            Map<List<? extends ThrowsTree>, ExecutableElement> declaredExceptionTags = new LinkedHashMap<>();
-            for (TypeMirror declaredExceptionType : declaredExceptionTypes) {
-                Input input = new DocFinder.Input(utils, holder, this,
-                        utils.getTypeName(declaredExceptionType, false));
-                DocFinder.Output inheritedDoc = DocFinder.search(writer.configuration(), input);
-                if (inheritedDoc.tagList.isEmpty()) {
-                    String typeName = utils.getTypeName(declaredExceptionType, true);
-                    input = new DocFinder.Input(utils, holder, this, typeName);
-                    inheritedDoc = DocFinder.search(writer.configuration(), input);
-                }
-                if (!inheritedDoc.tagList.isEmpty()) {
-                    if (inheritedDoc.holder == null) {
-                        inheritedDoc.holder = holder;
-                    }
-                    List<? extends ThrowsTree> inheritedTags = inheritedDoc.tagList.stream()
-                            .map(t -> (ThrowsTree) t)
-                            .toList();
-                    declaredExceptionTags.put(inheritedTags, (ExecutableElement) inheritedDoc.holder);
+        if (holder.getKind() != ElementKind.METHOD) {
+            // (Optimization.)
+            // Of all executable elements, only methods and constructors are documented.
+            // Of these two, only methods inherit documentation.
+            // Don't waste time on constructors.
+            assert holder.getKind() == ElementKind.CONSTRUCTOR : holder.getKind();
```

Comment:
This is weaker than the old code, which would throw an Error on other types.  `assert` may or may not be enabled.

Reply:
Hm... Which kind of "Error" would be thrown for which "other type"? I don't think that this method is ever called with an element other than a constructor or method; look:

    public ThrowsTaglet() {
        super(DocTree.Kind.THROWS, false, EnumSet.of(Location.CONSTRUCTOR, Location.METHOD));
    }

It is called for constructors regularly, but as you might imagine, it has no effect other than waste of resources and, what is more important, loss of code clarity. There are multiple types of executable elements. It's both clarifying and reassuring to see an assertion that establishes our programming assumptions as to which of those types we expect there, early in the method.

Reply:
I continue to think that `assert` is little better than a stylized comment, since they are typically not enabled.